Examples/Swift/CustomViewController.swift

```diff
@@ -0,0 +1,195 @@
+        let distance = routeProgress.currentLegProgress.currentStepProgress.distanceRemaining
+        self.distanceLabel.text = lengthFormatter.string(fromMeters: distance)
+    }
+
+    // Fired when the user is no longer on the route.
+    // A new route should be fetched at this time.
+    func rerouted(_ notification: NSNotification) {
+        speechSynth.stopSpeaking(at: .word)
+
+        getRoute {
+            /*
+             **IMPORTANT**
+
+             When rerouting, you need to give the RouteController a new route.
+             Otherwise, it will continue to compare the user to the old route and continually reroute the user.
+             */
+            let routeProgress = RouteProgress(route: self.userRoute!)
+            self.routeController.routeProgress = routeProgress
+            self.updateRouteProgress(routeProgress: routeProgress)
+        }
+    }
+
+    func getRoute(completion: (()->())? = nil) {
+        let options = RouteOptions(coordinates: [mapView.userLocation!.coordinate, destination.coordinate])
+        options.includesSteps = true
+        options.routeShapeResolution = .full
+        options.profileIdentifier = .automobileAvoidingTraffic
+
+        _ = directions.calculate(options) { [weak self] (waypoints, routes, error) in
+            guard error == nil else {
+                print(error!)
+                return
+            }
+
+            guard let route = routes?.first else {
+                return
+            }
+
+            self?.userRoute = route
+
+            completion?()
+            self?.addRouteToMap()
+        }
+    }
+
+    func addRouteToMap() {
+        guard let style = mapView.style else { return }
+        guard let userRoute = userRoute else { return }
+
+        mapView.addAnnotation(destination)
+
+        let polyline = MGLPolylineFeature(coordinates: userRoute.coordinates!, count: userRoute.coordinateCount)
+        let lineSource = MGLShapeSource(identifier: sourceIdentifier, shape: polyline, options: nil)
+
+        if let source = style.source(withIdentifier: sourceIdentifier) as? MGLShapeSource {
+            source.shape = polyline
+        } else {
+            let line = MGLLineStyleLayer(identifier: layerIdentifier, source: lineSource)
+
+            // Style the line
+            line.lineColor = MGLStyleValue(rawValue: UIColor(red:0.00, green:0.45, blue:0.74, alpha:0.9))
+            line.lineWidth = MGLStyleValue(rawValue: 5)
+            line.lineCap = MGLStyleValue(rawValue: NSValue(mglLineCap: .round))
+            line.lineJoin = MGLStyleValue(rawValue: NSValue(mglLineJoin: .round))
+
+            // Add source and layer
+            style.addSource(lineSource)
+            for layer in style.layers.reversed() {
+                if !(layer is MGLSymbolStyleLayer) {
+                    style.insertLayer(line, above: layer)
+                    break
+                }
+            }
+        }
+    }
+
+    func speak(_ text: String) {
```

Comment:
Since this is only used in one place, I think we can remove this function and move it up to where `speak()` is called.